test/functional/feature_rbf.py

```diff
@@ -60,14 +61,12 @@ def run_test(self):
         self.log.info("Running test spends of conflicting outputs...")
         self.test_spends_of_conflicting_outputs()
 
-        self.log.info("Running test new unconfirmed inputs...")
-        self.test_new_unconfirmed_inputs()
+        # TODO: rework too many replacements test to use direct conflicts only
+        #self.log.info("Running test too many replacements...")
+        #self.test_too_many_replacements()
 
-        self.log.info("Running test too many replacements...")
-        self.test_too_many_replacements()
-
-        self.log.info("Running test too many replacements using default mempool params...")
-        self.test_too_many_replacements_with_default_mempool_params()
+        #self.log.info("Running test too many replacements using default mempool params...")
+        #self.test_too_many_replacements_with_default_mempool_params()
```

Comment:
d0576c134a5
I think we only need to adapt 1 of these 2 tests and it should be quite easy to do. Currently, `test_too_many_replacements` requires a high descendant limit because it's 1 parent with 100 children, but we can just add a `self.generate(node, 1)` in between and it can use node 1 (the default one). Then, we can just delete `test_too_many_replacements_with_default_mempool_params`

```
diff --git a/test/functional/feature_rbf.py b/test/functional/feature_rbf.py
index f1de04008c4..57817380c50 100755
--- a/test/functional/feature_rbf.py
+++ b/test/functional/feature_rbf.py
@@ -61,12 +61,8 @@ class ReplaceByFeeTest(BitcoinTestFramework):
         self.log.info("Running test spends of conflicting outputs...")
         self.test_spends_of_conflicting_outputs()
 
-        # TODO: rework too many replacements test to use direct conflicts only
-        #self.log.info("Running test too many replacements...")
-        #self.test_too_many_replacements()
-
-        #self.log.info("Running test too many replacements using default mempool params...")
-        #self.test_too_many_replacements_with_default_mempool_params()
+        self.log.info("Running test too many replacements...")
+        self.test_too_many_replacements()
 
         self.log.info("Running test opt-in...")
         self.test_opt_in()
@@ -331,17 +327,20 @@ class ReplaceByFeeTest(BitcoinTestFramework):
         split_value = int((initial_nValue - fee) / (MAX_REPLACEMENT_LIMIT + 1))
 
         splitting_tx_utxos = self.wallet.send_self_transfer_multi(
-            from_node=self.nodes[0],
+            from_node=self.nodes[1],
             utxos_to_spend=[utxo],
             sequence=0,
             num_outputs=MAX_REPLACEMENT_LIMIT + 1,
             amount_per_output=split_value,
         )["new_utxos"]
 
+        # Let this tx confirm so that its children are not connected
+        self.generate(self.nodes[1], 1)
+
         # Now spend each of those outputs individually
         for utxo in splitting_tx_utxos:
             self.wallet.send_self_transfer(
-                from_node=self.nodes[0],
+                from_node=self.nodes[1],
                 utxo_to_spend=utxo,
                 sequence=0,
                 fee=Decimal(fee) / COIN,
@@ -359,98 +358,13 @@ class ReplaceByFeeTest(BitcoinTestFramework):
         double_tx_hex = double_tx.serialize().hex()
 
         # This will raise an exception
-        assert_raises_rpc_error(-26, "too many potential replacements", self.nodes[0].sendrawtransaction, double_tx_hex, 0)
+        assert_raises_rpc_error(-26, "too many potential replacements", self.nodes[1].sendrawtransaction, double_tx_hex, 0)
 
         # If we remove an input, it should pass
         double_tx.vin.pop()
         double_tx_hex = double_tx.serialize().hex()
         self.nodes[0].sendrawtransaction(double_tx_hex, 0)
 
-    def test_too_many_replacements_with_default_mempool_params(self):
-        """
-        Test rule 5 (do not allow replacements that cause more than 100
-        evictions) without having to rely on non-default mempool parameters.
-
-        In order to do this, create a number of "root" UTXOs, and then hang
-        enough transactions off of each root UTXO to exceed the MAX_REPLACEMENT_LIMIT.
-        Then create a conflicting RBF replacement transaction.
-        """
-        # Clear mempools to avoid cross-node sync failure.
-        for node in self.nodes:
-            self.generate(node, 1)
-        normal_node = self.nodes[1]
-        wallet = MiniWallet(normal_node)
-
-        # This has to be chosen so that the total number of transactions can exceed
-        # MAX_REPLACEMENT_LIMIT without having any one tx graph run into the descendant
-        # limit; 10 works.
-        num_tx_graphs = 10
-
-        # (Number of transactions per graph, rule 5 failure expected)
-        cases = [
-            # Test the base case of evicting fewer than MAX_REPLACEMENT_LIMIT
-            # transactions.
-            ((MAX_REPLACEMENT_LIMIT // num_tx_graphs) - 1, False),
-
-            # Test hitting the rule 5 eviction limit.
-            (MAX_REPLACEMENT_LIMIT // num_tx_graphs, True),
-        ]
-
-        for (txs_per_graph, failure_expected) in cases:
-            self.log.debug(f"txs_per_graph: {txs_per_graph}, failure: {failure_expected}")
-            # "Root" utxos of each txn graph that we will attempt to double-spend with
-            # an RBF replacement.
-            root_utxos = []
-
-            # For each root UTXO, create a package that contains the spend of that
-            # UTXO and `txs_per_graph` children tx.
-            for graph_num in range(num_tx_graphs):
-                root_utxos.append(wallet.get_utxo())
-
-                optin_parent_tx = wallet.send_self_transfer_multi(
-                    from_node=normal_node,
-                    sequence=MAX_BIP125_RBF_SEQUENCE,
-                    utxos_to_spend=[root_utxos[graph_num]],
-                    num_outputs=txs_per_graph,
-                )
-                assert_equal(True, normal_node.getmempoolentry(optin_parent_tx['txid'])['bip125-replaceable'])
-                new_utxos = optin_parent_tx['new_utxos']
-
-                for utxo in new_utxos:
-                    # Create spends for each output from the "root" of this graph.
-                    child_tx = wallet.send_self_transfer(
-                        from_node=normal_node,
-                        utxo_to_spend=utxo,
-                    )
-
-                    assert normal_node.getmempoolentry(child_tx['txid'])
-
-            num_txs_invalidated = len(root_utxos) + (num_tx_graphs * txs_per_graph)
-
-            if failure_expected:
-                assert num_txs_invalidated > MAX_REPLACEMENT_LIMIT
-            else:
-                assert num_txs_invalidated <= MAX_REPLACEMENT_LIMIT
-
-            # Now attempt to submit a tx that double-spends all the root tx inputs, which
-            # would invalidate `num_txs_invalidated` transactions.
-            tx_hex = wallet.create_self_transfer_multi(
-                utxos_to_spend=root_utxos,
-                fee_per_output=10_000_000,  # absurdly high feerate
-            )["hex"]
-
-            if failure_expected:
-                assert_raises_rpc_error(
-                    -26, "too many potential replacements", normal_node.sendrawtransaction, tx_hex, 0)
-            else:
-                txid = normal_node.sendrawtransaction(tx_hex, 0)
-                assert normal_node.getmempoolentry(txid)
-
-        # Clear the mempool once finished, and rescan the other nodes' wallet
-        # to account for the spends we've made on `normal_node`.
-        self.generate(normal_node, 1)
-        self.wallet.rescan_utxos()
-
     def test_opt_in(self):
         """Replacing should only work if orig tx opted in"""
         tx0_outpoint = self.make_utxo(self.nodes[0], int(1.1 * COIN))
```

Reply:
Thanks, took this.